### ISSUE TITLE
QR tests are failing because of tolerance

### ISSUE THREAD

**Issue by bartvm** [torch/torch7#376]
On two separate machines (but both using OpenBLAS) some the QR tests fail when using single precision, all falling short of the given tolerance by a small margin: https://gist.github.com/bartvm/90b11759edbfe23d796f

**Comment by soumith** [comment]
@bartvm should I just bump up the tolerance?

**Comment by bartvm** [comment]
Yeah, I guess that the tolerance is just slightly too high for some BLAS libraries. Looks like `2e-05` would do the trick already.

**Comment by andreaskoepf** [comment]
Shouldn't we have deterministic unit tests (I think @hughperkins already mentioned this)? This could be achieved by explicitly setting the rng seed value at the beginning of each unit test. If we want to do some extensive testing to include some unlikely constellations we could create a procedure that runs each test 100 or 1000 times with different (but still 100% reproducible) seed-values.
